src/tracer/MOM_generic_tracer.F90

```diff
@@ -584,13 +584,24 @@ subroutine MOM_generic_tracer_column_physics(h_old, h_new, ea, eb, fluxes, Hml,
                optics%nbands, optics%max_wavelength_band, optics%sw_pen_band, optics%opacity_band, &
                internal_heat=tv%internal_heat, frunoff=fluxes%frunoff, sosga=sosga, geolat=G%geolatT, eqn_of_state=tv%eqn_of_state)
```

Comment:
@andrew-c-ross, Thanks for opening this PR. While I think the associated [ocean BGC PR](https://github.com/NOAA-CEFI-Regional-Ocean-Modeling/ocean_BGC/pull/86) should operate effectively on its own, it is good to implement an additional safety check within MOM6. Do you think we should extend this check to lines 582-686 as well?

Reply:
Thanks for testing it out, Yi-Cheng. My understanding is that we don't need a similar check on lines 582--585 because there, regardless of whether `tv%internal_heat` is null or not, it will just be passed along to `generic_tracer_source`, where a null pointer will then be caught by the `if(present())` check. The check that I did add when units scaling is being done is necessary because the scaling does `G%US%RZ_to_kg_m2*US%C_to_degC*tv%internal_heat(:,:)` before passing to `generic_tracer_source`, and that will segfault if internal_heat is null.